Build failure: dura

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *dura*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>
<pre><code>
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/lld169gm5c4ph7cyq5mdvsqdfb2n2vn0-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/mnmyz329jx52x4gmnxfb3wha6g1pxrfs-dura-0.2.0-vendor.tar.gz
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/qazsgkw6jgwkysmavg3glfpflj05c2zk-Cargo.lock.patch
patching file Cargo.lock
Executing cargoSetupPostPatchHook
Validating consistency between /private/tmp/nix-build-dura-0.2.0.drv-0/source/Cargo.lock and /private/tmp/nix-build-dura-0.2.0.drv-0/dura-0.2.0-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing cargoBuildHook
cargoBuildHook flags: -j 12 --target aarch64-apple-darwin --offline --profile release
   Compiling libc v0.2.112
   Compiling autocfg v1.0.1
   Compiling proc-macro2 v1.0.36
   Compiling pkg-config v0.3.24
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.85
   Compiling cfg-if v1.0.0
   Compiling log v0.4.14
   Compiling lazy_static v1.4.0
   Compiling memchr v2.4.1
   Compiling tinyvec_macros v0.1.0
   Compiling serde_derive v1.0.133
   Compiling tinyvec v1.5.1
   Compiling serde v1.0.133
   Compiling matches v0.1.9
   Compiling regex-syntax v0.6.25
   Compiling smallvec v1.7.0
   Compiling num-traits v0.2.14
   Compiling parking_lot_core v0.8.5
   Compiling num-integer v0.1.44
   Compiling indexmap v1.8.0
   Compiling tracing-core v0.1.21
   Compiling quote v1.0.14
   Compiling unicode-normalization v0.1.19
   Compiling instant v0.1.12
   Compiling once_cell v1.9.0
   Compiling jobserver v0.1.24
   Compiling percent-encoding v2.1.0
   Compiling scopeguard v1.1.0
   Compiling cc v1.0.72
   Compiling unicode-bidi v0.3.7
   Compiling pin-project-lite v0.2.8
   Compiling lock_api v0.4.5
   Compiling form_urlencoded v1.0.1
   Compiling regex-automata v0.1.10
   Compiling idna v0.2.3
   Compiling serde_json v1.0.74
   Compiling bitflags v1.3.2
   Compiling hashbrown v0.11.2
   Compiling url v2.2.2
   Compiling parking_lot v0.11.2
   Compiling tracing-log v0.1.2
   Compiling openssl-sys v0.9.72
   Compiling libz-sys v1.1.3
   Compiling libssh2-sys v0.2.23
   Compiling libgit2-sys v0.12.26+1.3.0
   Compiling matchers v0.1.0
   Compiling mio v0.7.14
   Compiling regex v1.5.5
   Compiling os_str_bytes v6.0.0
   Compiling thread_local v1.1.4
   Compiling signal-hook-registry v1.4.0
   Compiling dirs-sys v0.3.6
   Compiling atty v0.2.14
   Compiling time v0.1.44
   Compiling num_cpus v1.13.1
   Compiling sharded-slab v0.1.4
   Compiling termcolor v1.1.2
   Compiling itoa v1.0.1
   Compiling ansi_term v0.12.1
   Compiling bytes v1.1.0
   Compiling ryu v1.0.9
   Compiling strsim v0.10.0
   Compiling textwrap v0.15.0
   Compiling clap v3.1.6
   Compiling chrono v0.4.19
   Compiling dirs v4.0.0
   Compiling tracing-attributes v0.1.18
   Compiling tokio-macros v1.7.0
The following warnings were emitted during compilation:
<omitted for brevity>

  --- stderr
  src/smoke.c:1:10: fatal error: 'zlib.h' file not found
      1 | #include <zlib.h>
        |          ^~~~~~~~
  1 error generated.


  error occurred: Command env -u IPHONEOS_DEPLOYMENT_TARGET "/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-arch" "arm64" "-I" "src/zlib" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-D_C99_SOURCE" "-o" "/private/tmp/nix-build-dura-0.2.0.drv-0/source/target/aarch64-apple-darwin/release/build/libz-sys-3098ebdcc4f223fc/out/lib/src/zlib/zutil.o" "-c" "src/zlib/zutil.c" with args "cc" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
</code></pre>

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@drupol

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/misc/dura/default.nix
+++ b/pkgs/development/tools/misc/dura/default.nix
@@ -6,6 +6,7 @@
   openssl,
   pkg-config,
   Security,
+  zlib,
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -30,6 +31,7 @@ rustPlatform.buildRustPackage rec {
   buildInputs =
     [
       openssl
+      zlib
     ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
       Security

